src/preferences/dialog/dlgprefcontrols.cpp

```diff
@@ -681,6 +701,30 @@ void DlgPrefControls::slotApply() {
 
     int configSPAutoReset = BaseTrackPlayer::RESET_NONE;
 
+    // screensaver mode update
+    int inhibitcombo = comboBoxScreensaver->itemData(
+            comboBoxScreensaver->currentIndex()).toInt();
+    int inhibitsettings = m_pConfig->getValue<int>(ConfigKey("[Config]","InhibitScreensaver"));
+    if (inhibitcombo != inhibitsettings) {
+        m_pConfig->set(ConfigKey("[Config]", "InhibitScreensaver"), ConfigValue(inhibitcombo));
+
+        mixxx::ScreenSaverPreference inhiOld = mixxx::ScreenSaverPreference(inhibitsettings);
+        if (inhiOld == mixxx::ScreenSaverPreference::PREVENT_ON) {
+            mixxx::ScreenSaverHelper::uninhibit();
+        } else if (inhiOld == mixxx::ScreenSaverPreference::PREVENT_ON_PLAY) {
+            mixxx::ScreenSaverHelper::inhibitOnCondition(false);
```

Comment:
Conceptually yes, but in practice no.
Some of the implementations require that you call the pair inhibit as many times as uninhibit.  (as in, calling inhibit twice and unhinibit once would not uninhibit. So just in case I try to keep that).

I probably need to find a better name for this method. What it really does is either inhibit or uninhibit, depending if the parameter is the same or different than the current status. like this:

Current status: inhibit on
inhibitOnCondition(true) -> does nothing
inhibitOnCondition(false) -> unhinibit.
Current status: inhibit off
inhibitOnCondition(true) -> inhibit
inhibitOnCondition(false) -> does nothing

The end result is always that true leaves it enabled and false leaves it disabled, but internally it decides if it has to change anything or not. 

But now that I think about it, I'm not sure why i needed it as separate method. Will check and comment later.